pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -94,6 +98,37 @@ func (hs *proxierHealthServer) QueuedUpdate() {
 	hs.oldestPendingQueued.CompareAndSwap(zeroTime, hs.clock.Now())
 }
 
+// IsHealthy returns the proxier's health state, following the same definition
+// the HTTP server defines.
+func (hs *proxierHealthServer) IsHealthy() bool {
+	isHealthy, _, _ := hs.isHealthy()
+	return isHealthy
+}
+
+func (hs *proxierHealthServer) isHealthy() (bool, time.Time, time.Time) {
```

Comment:
just a thought, we are returning the two `time.Time` just for the output message, why if we just output the message directly?

```
// isHealthy returns the state of the proxy and a message with the reason in case is not healthy
func (hs *proxierHealthServer) isHealthy() (bool, string)
...
return false, fmt.Printf(`{"lastUpdated": %q,"currentTime": %q}`, lastUpdated, currentTime)
```

or just an error with the message

```
// isHealthy returns and error if not healthy
func (hs *proxierHealthServer) isHealthy() error
...
return fmt.Errorf(`{"lastUpdated": %q,"currentTime": %q}`, lastUpdated, currentTime)
```

Reply:
I will let owners / people more experienced with the proxy code comment on this.

Reply:
I would not rely on an `error.Error()` to return valid JSON - that's a little weird.

We could return the JSON as `[]byte` or even return a struct but I don't think it's worth the extra plumbing.